include/index_factory.h

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "index.h"
+#include "common_includes.h"
+
+namespace diskann
+{
+template <typename T, typename TagT, typename LabelT> class MemoryIndex;
+template <typename T, typename LabelT> class DiskIndex;
+enum BuildType
+{
+    MEMORY,
```

Comment:
I dont think MEMORY/DISK is the appropriate hierarchy here. The index consists of multiple objects such as 1 or 2 data stores, graph stores, filter stores. Each of those objects can be on disk, memory, etc. So the config is the choice of data and graph stores.

Reply:
ok, so the primary choice should be load_store_stratagy. But that part is being provided in the index.cpp via  a factory. In order to achieve that we can just pass an arg into the index constructor, and possibly initiate an index constructor by a config. So what would be the point of having an index factory? That part I am missing a bit.